CRM/Core/Smarty.php

```diff
@@ -158,6 +158,17 @@ private function initialize() {
     }
   }
 
+  /**
+   * Is this a test environment.
+   *
+   * Currently we just check the class name but it is possible to run tests in CMS frameworks.
+   *
+   * @return bool
+   */
+  private static function isTestEnvironment(): bool {
+    return CRM_Core_Config::singleton()->userFramework === 'CRM_Utils_System_UnitTests';
```

Comment:
Is this right? Would it just be 'UnitTests'?

Reply:
Also bootstrap [defines CIVICRM_TEST](https://github.com/civicrm/civicrm-core/blob/8104b62cb9b474aae6520139b02dd98d2eb9dc53/tests/phpunit/CiviTest/bootstrap.php#L17) and I've seen that used elsewhere.